lib/auth.js

```diff
@@ -0,0 +1,102 @@
+/* @auth.js
+ *
+ * Authuthentication functions
+ */
+
+const { instructorOwnsCourse, studentInCourse } = require('../models/course');
+const CustomError = require("../lib/custom-error");
+
+const jwt = require('jsonwebtoken');
+const secretKey = 'SuperSecret!';
+
+/*
+ * Generate a JWT Token
+ */
+exports.generateAuthToken = (user) => {
+  const payload = {
+    sub: user.id,
+    email: user.email,
```

Comment:
maybe remove user.email and user.name and just keep user.id?

Reply:
Yeah we can debate what we need in the JWT. I just put it all in there for now, it would be nice to have enough data in the JWT so we don't need to execute a DB lookup.